channels/rdp2tcp/client/rdp2tcp_main.c

```diff
@@ -168,7 +171,9 @@ static DWORD WINAPI copyThread(void* data)
 			goto fail;
 		}
 
-		WaitForSingleObject(plugin->writeComplete, INFINITE);
+		handles[0] = plugin->writeComplete;
+		handles[1] = freerdp_abort_event(plugin->channelEntryPoints.context);
+		status = WaitForMultipleObjects(2, handles, FALSE, INFINITE);
 		ResetEvent(plugin->writeComplete);
```

Comment:
Here we would ResetEvent `plugin->writeComplete` unconditionnally even if status is an error, that seems fishy

Reply:
@hardening well, as the other condition is `terminate the session` I didn´t think the check would be useful, but yes, could be added.

Reply:
but fixed nevertheless.